app/command_mute.go

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+package app
+
+import (
+	"strings"
+
+	"github.com/mattermost/mattermost-server/model"
+	goi18n "github.com/nicksnyder/go-i18n/i18n"
+)
+
+type MuteProvider struct {
+}
+
+const (
+	CMD_MUTE = "mute"
+)
+
+func init() {
+	RegisterCommandProvider(&MuteProvider{})
+}
+
+func (me *MuteProvider) GetTrigger() string {
+	return CMD_MUTE
+}
+
+func (me *MuteProvider) GetCommand(a *App, T goi18n.TranslateFunc) *model.Command {
+	return &model.Command{
+		Trigger:          CMD_MUTE,
+		AutoComplete:     true,
+		AutoCompleteDesc: T("api.command_mute.desc"),
+		AutoCompleteHint: T("api.command_mute.hint"),
+		DisplayName:      T("api.command_mute.name"),
+	}
+}
+
+func (me *MuteProvider) DoCommand(a *App, args *model.CommandArgs, message string) *model.CommandResponse {
+	var channel *model.Channel
+	var noChannelErr *model.AppError
+
+	if channel, noChannelErr = a.GetChannel(args.ChannelId); noChannelErr != nil {
+		return &model.CommandResponse{Text: args.T("api.command_mute.error", map[string]interface{}{"Channel": channel.DisplayName}), ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL}
```

Comment:
This is correct? If there is an error, the channel will be nil, or am I missing something?

Reply:
Hej @cpanato,

I think you're right. I'll fix this this afternoon (CET). Very thoughtful :)

Reply:
Hej, see https://github.com/mattermost/mattermost-server/pull/8575